apps/src/code-studio/pd/form_components/FormController.jsx

```diff
@@ -392,9 +392,9 @@ export default class FormController extends React.Component {
       this.setState({
         currentPage: newPage
       });
-    }
 
-    this.saveToSessionStorage({currentPage: newPage});
+      this.saveToSessionStorage({currentPage: newPage});
```

Comment:
This is a bug I discovered in testing the email validator. We should only persist the new page when the current one is valid and we actually navigate.